hedera-node/docs/design/services/smart-contract-service/RedirectContract-description.md

```diff
@@ -0,0 +1,263 @@
+   return the customized redirect contract for this account address; otherwise return the empty
+   bytecode (which lets the EVM do the "successful noop" call of a nonexistent contract)
+   * _n.b._: "_and_ the account id given exists": Only for HAS and HSS at this time, not yet for HTS.
+4. `DispatchingEvmFrameState.getAccountRedirectCode` - given account id return the bytecode
+   for the redirect bytecode specialized for this account
+5. From this point it behaves like any other contract call, and the bytecode is executed
+6. In that bytecode it will do a delegate call to the HAS system contract, passing along the
+   original calldata, and return the result returned from the system contract.  Thus: It is
+   behaves as a proxy/intermediary contract shimming the call to an _account address_ to a call
+   to the _system contract address_.
+   * _n.b.:_ This is the _sole_ exception to the Smart Contract Security Model v2 rule that
+     prohibits a _delegate_ call to a system contract method:  If it is done by the system by using
+     its redirect contract.  (See `FrameUtils.callTypeOf()`.)
+
+## Other decompiles/disassemblies of the redirect contract
+
+### 3-address
+
+```
+function __function_selector__() public {
+    Begin block 0x0
+    prev=[], succ=[0xb, 0xf]
+    =================================
+    0x0: v0(0x80) = CONST
+    0x2: v2(0x40) = CONST
+    0x4: MSTORE v2(0x40), v0(0x80)
+    0x5: v5 = CALLVALUE
+    0x7: v7 = ISZERO v5
+    0x8: v8(0xf) = CONST
+    0xa: JUMPI v8(0xf), v7
+
+    Begin block 0xb
+    prev=[0x0], succ=[]
+    =================================
+    0xb: vb(0x0) = CONST
+    0xe: REVERT vb(0x0), vb(0x0)
+
+    Begin block 0xf
+    prev=[0x0], succ=[0x58, 0x54]
+    =================================
+    0x11: v11(0x0) = CONST
+    0x13: v13(0x167) = CONST
+    0x18: v18(0x618dc65efefefefefefefefefefefefefefefefefefefefe) = CONST
+    0x31: v31(0x0) = CONST
+    0x33: MSTORE v31(0x0), v18(0x618dc65efefefefefefefefefefefefefefefefefefefefe)
+    0x34: v34 = CALLDATASIZE
+    0x35: v35(0x0) = CONST
+    0x37: v37(0x20) = CONST
+    0x39: CALLDATACOPY v37(0x20), v35(0x0), v34
+    0x3a: v3a(0x0) = CONST
+    0x3d: v3d = CALLDATASIZE
+    0x3e: v3e(0x18) = CONST
+    0x40: v40 = ADD v3e(0x18), v3d
+    0x41: v41(0x8) = CONST
+    0x44: v44 = GAS
+    0x45: v45 = DELEGATECALL v44, v13(0x167), v41(0x8), v40, v3a(0x0), v3a(0x0)
+    0x46: v46 = RETURNDATASIZE
+    0x48: v48(0x0) = CONST
+    0x4b: RETURNDATACOPY v48(0x0), v48(0x0), v46
+    0x4d: v4d(0x0) = CONST
+    0x50: v50 = EQ v45, v4d(0x0)
+    0x51: v51(0x58) = CONST
+    0x53: JUMPI v51(0x58), v50
+
+    Begin block 0x58
+    prev=[0xf], succ=[]
+    =================================
+    0x5a: v5a(0x0) = CONST
+    0x5c: REVERT v5a(0x0), v46
+
+    Begin block 0x54
+    prev=[0xf], succ=[]
+    =================================
+    0x55: v55(0x0) = CONST
+    0x57: RETURN v55(0x0), v46
+
+}
+```
+
+### Disassembly
+
+Via `library.dedaub.com/decompile`:
```

Comment:
```suggestion
Via `library.dedaub.com/disassembly`:
```
??

Reply:
`library.dedaub.com/disassembly` gives 404 - it was `/decompile` I used ...